### PR TITLE
Add section on issue and pull request labels to contributing guide

### PR DESCRIPTION
**Pull request by izuzak**
This adds a section on labels used in `atom/atom` and across the Atom organization, something that the @atom/issue-triage team discussed previously.

Notes:
- The diff is rather large due to all the search links, it might be easier to read in rendered mode
- The intro to the section is a bit wordy as I tried to explain the motivation for listing these labels and adding the search links. I'm not worried about that, and I think the words are worth it, but if someone has suggestions on reducing that text -- that'd be swell.
- Also wanted to make it clear that these labels are not set in stone and that we want to evolve them to support the project and the community. 
- I added this section at the very bottom of the document so that it doesn't overwhelm readers who are reading top-to-bottom.
- Also added a TOC section at the top to announce what the guide talks about and make it easier to jump to specific sections.
-  I think I listed all labels that are out there, but please let me know if I missed one. Also, I wonder if some labels shouldn't be listed for some reason (I can't think of such a reason, but still wondering if someone else can).

@atom/issue-triage When you have some time to :eyes: this :books:, would love to hear your thoughts on the format and the content. :thought_balloon: Also /cc @atom/core.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+
+The labels are loosely grouped by their purpose, but it's not required that every issue has a label from every group or that an issue can't have more than one label from the same group.
+
+Please open an issue if you have suggestions for new labels or if you notice some labels are missing on some repositories.
```
Comment (lee-dohm) [inline CONTRIBUTING.md]: Where should one open an Issue? Should they all just be opened on atom/atom?

Should we have a `label` label? :laughing:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+
+This section lists the labels we use to help us track and manage issues and pull request. Most labels are used across the core `atom/atom` repository and all package repositories, but some are specific for `atom/atom`.
+
+[GitHub  search](https://help.github.com/articles/searching-issues/) makes it easy to use labels for finding groups of issues or pull requests you're interested in. For example, you might be interested in [open issues across `atom/atom` and all Atom-owned packages which are labeled as bugs, but still need to be reliably reproduced](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+user%3Aatom+label%3Abug+label%3Aneeds-reproduction) or perhaps [open pull requests in `atom/atom` which haven't been reviewed yet](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+repo%3Aatom%2Fatom+comments%3A0). To help you find issues and pull requests, each label is listed with search links for finding open items with that label in `atom/atom` only and also across all Atom repositories. We  encourage you to read about [other search filters](https://help.github.com/articles/searching-issues/) which will help you write more focused queries.
```
Comment (mnquintana) [inline CONTRIBUTING.md]: Super minor, but there's an extra space between `GitHub` and `search` :art:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
 # Returns `undefined`.
 disablePackage: (name, options, callback) ->
 ```
+
+## Issue and Pull Request Labels
+
+This section lists the labels we use to help us track and manage issues and pull request. Most labels are used across the core `atom/atom` repository and all package repositories, but some are specific for `atom/atom`.
```
Comment (mnquintana) [inline CONTRIBUTING.md]: `pull request` -> `pull requests`?

**Comment by izuzak** [comment]
Thanks for the suggestions, @lee-dohm and @mnquintana -- just pushed https://github.com/atom/atom/compare/0cd007dfb18d...7ea71a94a547 with some fixes.

I'll leave this open for a few more days in case other members of @atom/issue-triage or @atom/core have any comments. :speech_balloon:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+
+[GitHub search](https://help.github.com/articles/searching-issues/) makes it easy to use labels for finding groups of issues or pull requests you're interested in. For example, you might be interested in [open issues across `atom/atom` and all Atom-owned packages which are labeled as bugs, but still need to be reliably reproduced](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+user%3Aatom+label%3Abug+label%3Aneeds-reproduction) or perhaps [open pull requests in `atom/atom` which haven't been reviewed yet](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+repo%3Aatom%2Fatom+comments%3A0). To help you find issues and pull requests, each label is listed with search links for finding open items with that label in `atom/atom` only and also across all Atom repositories. We  encourage you to read about [other search filters](https://help.github.com/articles/searching-issues/) which will help you write more focused queries.
+
+The labels are loosely grouped by their purpose, but it's not required that every issue has a label from every group or that an issue can't have more than one label from the same group.
```
Comment (mnquintana) [inline CONTRIBUTING.md]: :art: `that every issue has a label` -> `that every issue have a label`?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+
+#### Core Team Project Management
+
+| Label name | `atom/atom` search | `atom`‑org :mag_right: | Description |
+| --- | --- | --- | --- |
+| `in-progress` | [search][search-atom-repo-label-in-progress] | [search][search-atom-org-label-in-progress] | Tasks which the Atom core team is working on currently. |
+| `on-deck` | [search][search-atom-repo-label-on-deck] | [search][search-atom-org-label-on-deck] | Tasks which the Atom core team plans to work on next. |
+| `shipping` | [search][search-atom-repo-label-shipping] | [search][search-atom-org-label-shipping] | Tasks which the Atom core team completed and will be released in one of the next releases. |
+| `post-1.0-roadmap` | [search][search-atom-repo-label-post-1.0-roadmap] | [search][search-atom-org-label-post-1.0-roadmap] | Atom core team's roadmap post 1.0.0 version.  |
+| `atom` | [search][search-atom-repo-label-atom] | [search][search-atom-org-label-atom] | Topics discussed  for prioritization at next meeting of Atom core team members. |
```
Comment (mnquintana) [inline CONTRIBUTING.md]: :art: I think there's an extra space between `discussed` and `for`

Comment (mnquintana) [inline CONTRIBUTING.md]: `at next meeting` -> `at the next meeting`?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+
+#### Pull Request Labels
+
+| Label name | `atom/atom` pull requests | `atom`‑org pull requests | Description
+| --- | --- | --- | --- |
+| `work-in-progress` | [search][search-atom-repo-label-work-in-progress] | [search][search-atom-org-label-work-in-progress] | The author is still working on the pull request, more changes will follow. |
+| `needs-review` | [search][search-atom-repo-label-needs-review] | [search][search-atom-org-label-needs-review] | Needs code review, and also approval from maintainers or Atom core team. |
+| `under-review` | [search][search-atom-repo-label-under-review] | [search][search-atom-org-label-under-review] | Pull request is being reviewed by a maintainer or Atom core team. |
```
Comment (mnquintana) [inline CONTRIBUTING.md]: `or Atom core team member` maybe?

**Comment by mnquintana** [comment]
Other than my comments, this looks great to me! :rocket: :ship:

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
 # Returns `undefined`.
 disablePackage: (name, options, callback) ->
 ```
+
+## Issue and Pull Request Labels
+
+This section lists the labels we use to help us track and manage issues and pull requests. Most labels are used across the core `atom/atom` repository and all package repositories, but some are specific for `atom/atom`.
```
Comment (50Wliu) [inline CONTRIBUTING.md]: How about `Most labels are used across all atom repositories, but some are specific to atom/atom`?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+The labels are loosely grouped by their purpose, but it's not required that every issue has a label from every group or that an issue can't have more than one label from the same group.
+
+Please open an issue on `atom/atom` if you have suggestions for new labels, and if you notice some labels are missing on some repositories, then please open an issue on that repository.
+
+#### Type of Issue and Issue State
+
+| Label name | `atom/atom` :mag_right: | `atom`‑org :mag_right: | Description |
+| --- | --- | --- | --- |
+| `enhancement` | [search][search-atom-repo-label-enhancement] | [search][search-atom-org-label-enhancement] | A feature request. |
+| `bug` | [search][search-atom-repo-label-bug] | [search][search-atom-org-label-bug] | A confirmed bug or issue very likely to be a bug. |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: `[...]or something that is very likely to be a bug`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `question` | [search][search-atom-repo-label-question] | [search][search-atom-org-label-question] | A question more than a bug report or feature request (e.g. how do I do X). |
+| `feedback` | [search][search-atom-repo-label-feedback] | [search][search-atom-org-label-feedback] | General feedback more than a bug report or feature request. |
+| `help-wanted` | [search][search-atom-repo-label-help-wanted] | [search][search-atom-org-label-help-wanted] | The Atom core team would appreciate help from the community in resolving these issues. |
+| `beginner` | [search][search-atom-repo-label-beginner] | [search][search-atom-org-label-beginner] | Less complex issue which would be a good first issue to work on for users who want to contribute to Atom. |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: :art: issue -> issues
So `Less complex issues which would be good places to start for users[...]`?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `more-information-needed` | [search][search-atom-repo-label-more-information-needed] | [search][search-atom-org-label-more-information-needed] | More information needs to be collected about the problem or feature request (e.g. steps to reproduce). |
+| `needs-reproduction` | [search][search-atom-repo-label-needs-reproduction] | [search][search-atom-org-label-needs-reproduction] | Likely a bug, but hasn't been reliably reproduced. |
+| `triage-help-needed` | [search][search-atom-repo-label-triage-help-needed] | [search][search-atom-org-label-triage-help-needed] | Help is needed to triage this issue, normally from the Atom core team, e.g. to determine if it's a bug or not. |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: This label has never been used yet, so perhaps :fire:?

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `blocked` | [search][search-atom-repo-label-blocked] | [search][search-atom-org-label-blocked] | Issue is blocked on another issue. |
+| `duplicate` | [search][search-atom-repo-label-duplicate] | [search][search-atom-org-label-duplicate] | Issue is a duplicate of another issue. |
+| `wontfix` | [search][search-atom-repo-label-wontfix] | [search][search-atom-org-label-wontfix] | The Atom core team decided not to fix this issue for now, either because it's working as intended or for some other reason. |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: has decided

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `invalid` | [search][search-atom-repo-label-invalid] | [search][search-atom-org-label-invalid] | Issue isn't valid (e.g. user error). |
+| `package-idea` | [search][search-atom-repo-label-package-idea] | [search][search-atom-org-label-package-idea] | A feature request which might be a good candidate for a new package, instead of extending Atom or a core Atom package. |
+| `wrong-repo` | [search][search-atom-repo-label-wrong-repo] | [search][search-atom-org-label-wrong-repo] | Issue was reported on the wrong repository (e.g. a bug related to a package was reported on `atom/atom`). |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: It might be better to be more specific here, eg `a bug related to the settings view was reported on atom/atom`.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `mac` | [search][search-atom-repo-label-mac] | [search][search-atom-org-label-mac] | Related to Atom running on OSX. |
+| `documentation` | [search][search-atom-repo-label-documentation] | [search][search-atom-org-label-documentation] | Related to any type of documentation (e.g. [API documentation](https://atom.io/docs/api/latest/Atom) and the [flight manual](https://atom.io/docs/latest/)). |
+| `performance` | [search][search-atom-repo-label-performance] | [search][search-atom-org-label-performance] | Related to performance. |
+| `security` | [search][search-atom-repo-label-security] | [search][search-atom-org-label-security] | Related to security. |
+| `ui` | [search][search-atom-repo-label-ui] | [search][search-atom-org-label-ui] | Related to visual design. |
+| `api` | [search][search-atom-repo-label-api] | [search][search-atom-org-label-api] | Related to Atom's public APIs. |
+| `uncaught-exception` | [search][search-atom-repo-label-uncaught-exception] | [search][search-atom-org-label-uncaught-exception] | Issues created for uncaught exceptions thrown, normally from the [Notifications package](https://github.com/atom/notifications). |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: `Issues created for any uncaught exceptions, ...`?

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `crash` | [search][search-atom-repo-label-crash] | [search][search-atom-org-label-crash] | Problems which cause Atom to crash. |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: Problems-->Situations

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `error-from-pathwatcher` | [search][search-atom-repo-label-error-from-pathwatcher] | [search][search-atom-org-label-error-from-pathwatcher] | Related to errors thrown by the [pathwatcher library](https://github.com/atom/node-pathwatcher). |
+| `error-from-save` | [search][search-atom-repo-label-error-from-save] | [search][search-atom-org-label-error-from-save] | Related to errors thrown when saving files. |
+| `error-from-open` | [search][search-atom-repo-label-error-from-open] | [search][search-atom-org-label-error-from-open] | Related to errors thrown when opening files. |
+| `installer` | [search][search-atom-repo-label-installer] | [search][search-atom-org-label-installer] | Related to the Atom installers for different OSes. |
+| `auto-updater` | [search][search-atom-repo-label-auto-updater] | [search][search-atom-org-label-auto-updater] | Related to the auto-updater for different OSes. |
+| `deprecation-help` | [search][search-atom-repo-label-deprecation-help] | [search][search-atom-org-label-deprecation-help] | Issues for helping package authors remove usage of deprecated APIs in packages. |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: This could also be a candidate for removal - it's only been used once.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+
+#### Core Team Project Management
+
+| Label name | `atom/atom` search | `atom`‑org :mag_right: | Description |
+| --- | --- | --- | --- |
+| `in-progress` | [search][search-atom-repo-label-in-progress] | [search][search-atom-org-label-in-progress] | Tasks which the Atom core team is working on currently. |
+| `on-deck` | [search][search-atom-repo-label-on-deck] | [search][search-atom-org-label-on-deck] | Tasks which the Atom core team plans to work on next. |
+| `shipping` | [search][search-atom-repo-label-shipping] | [search][search-atom-org-label-shipping] | Tasks which the Atom core team completed and will be released in one of the next releases. |
+| `post-1.0-roadmap` | [search][search-atom-repo-label-post-1.0-roadmap] | [search][search-atom-org-label-post-1.0-roadmap] | Atom core team's roadmap post 1.0.0 version.  |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: :art: `The Atom core team's`

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| `error-from-save` | [search][search-atom-repo-label-error-from-save] | [search][search-atom-org-label-error-from-save] | Related to errors thrown when saving files. |
+| `error-from-open` | [search][search-atom-repo-label-error-from-open] | [search][search-atom-org-label-error-from-open] | Related to errors thrown when opening files. |
+| `installer` | [search][search-atom-repo-label-installer] | [search][search-atom-org-label-installer] | Related to the Atom installers for different OSes. |
+| `auto-updater` | [search][search-atom-repo-label-auto-updater] | [search][search-atom-org-label-auto-updater] | Related to the auto-updater for different OSes. |
+| `deprecation-help` | [search][search-atom-repo-label-deprecation-help] | [search][search-atom-org-label-deprecation-help] | Issues for helping package authors remove usage of deprecated APIs in packages. |
+
+#### Core Team Project Management
+
+| Label name | `atom/atom` search | `atom`‑org :mag_right: | Description |
```
Comment (50Wliu) [inline CONTRIBUTING.md]: atom/atom is missing `:mag_right:`

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -163,3 +173,180 @@ describe 'a dog', ->
...
+| --- | --- | --- | --- |
+| `in-progress` | [search][search-atom-repo-label-in-progress] | [search][search-atom-org-label-in-progress] | Tasks which the Atom core team is working on currently. |
+| `on-deck` | [search][search-atom-repo-label-on-deck] | [search][search-atom-org-label-on-deck] | Tasks which the Atom core team plans to work on next. |
+| `shipping` | [search][search-atom-repo-label-shipping] | [search][search-atom-org-label-shipping] | Tasks which the Atom core team completed and will be released in one of the next releases. |
+| `post-1.0-roadmap` | [search][search-atom-repo-label-post-1.0-roadmap] | [search][search-atom-org-label-post-1.0-roadmap] | Atom core team's roadmap post 1.0.0 version.  |
+| `atom` | [search][search-atom-repo-label-atom] | [search][search-atom-org-label-atom] | Topics discussed  for prioritization at next meeting of Atom core team members. |
+
+#### Pull Request Labels
+
+| Label name | `atom/atom` pull requests | `atom`‑org pull requests | Description
```
Comment (50Wliu) [inline CONTRIBUTING.md]: Both are missing `:mag_right:`

File: CONTRIBUTING.md (hunk 15 above)
Comment (izuzak) [inline CONTRIBUTING.md]: > This could also be a candidate for removal - it's only been used once.

Thanks for pointing this out, @50Wliu, and that `triage-help-needed` label. I removed the latter, but decided to keep this one for now, since it's been used once :one:. One of my motivations for documenting _everything_ (or as much of everything as possible :smile:) was a hope that users who read this might have ideas and suggest better labels/workflows for the project.

**Comment by izuzak** [comment]
Thanks for all the comments, @50Wliu and @mnquintana :hearts: -- I made those changes in 7a4a002. 

Since this has been open for two weeks, I'm merging it now. If you notice any other rough edges -- please feel free to smooth them out :gem:.
